### ISSUE TITLE
Error when using `executeStatement` to send a raw query.

### ISSUE THREAD

**Issue by ndawod** [j256/ormlite-core#303]
Hi there.

I'm using MySQL JDBC and wanting to temporarily disable foreign key checks with this command:

```java
final DatabaseConnection dc = connectionSource.getReadWriteConnection("")
dc.executeStatement("SET FOREIGN_KEY_CHECKS=0", DatabaseConnection.DEFAULT_RESULT_FLAGS)
```

However, I'm getting an error like this: `Statement.executeQuery() cannot issue statements that do not produce result sets.`

I did a bit of digging and the error is thrown in `JdbcCompiledStatement.java:64`:

```java
    public int runExecute() throws SQLException {
        if (!this.type.isOkForExecute()) {
            throw new IllegalArgumentException("Cannot call execute on a " + this.type + " statement");
        } else {
            this.preparedStatement.execute();
            return this.preparedStatement.getUpdateCount();
        }
    }
```

Then I traced back the `isOkForExecute` and saw the following:

![Screenshot 2024-10-22 at 10 49 57](https://github.com/user-attachments/assets/0e535769-0ed1-4320-b977-bf9773767409)

I see no match for queries starting with `SET …`, could this be related?

Bottom line: Given a `ConnectionSource`, how to send a raw query such as `SET FOREIGN_KEY_CHECKS=0`?

**Comment by j256** [comment]
Nw.  For posterity, you should have been using `dc.executeStatement()`, right?

**Comment by ndawod** [comment]
Correct, it was the right command :)
